custom/crediblex/portfolio/loc/src/main/java/com/crediblex/fineract/portfolio/loc/service/LineOfCreditWritePlatformServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+import org.apache.fineract.portfolio.client.domain.ClientRepositoryWrapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Slf4j
+public class LineOfCreditWritePlatformServiceImpl implements LineOfCreditWritePlatformService {
+
+    private final PlatformSecurityContext context;
+    private final LineOfCreditRepository lineOfCreditRepository;
+    private final ClientRepositoryWrapper clientRepository;
+    private final LineOfCreditDataValidator dataValidator;
+
+    @Autowired
+    public LineOfCreditWritePlatformServiceImpl(PlatformSecurityContext context, LineOfCreditRepository lineOfCreditRepository,
+            ClientRepositoryWrapper clientRepository, LineOfCreditDataValidator dataValidator) {
+        this.context = context;
+        this.lineOfCreditRepository = lineOfCreditRepository;
+        this.clientRepository = clientRepository;
+        this.dataValidator = dataValidator;
+    }
+
+    @Override
+    @Transactional
+    public CommandProcessingResult createLineOfCredit(JsonCommand command) {
+        try {
+            this.dataValidator.validateForCreate(command.json());
+
+            final Long clientId = command.longValueOfParameterNamed("clientId");
+
+            final Client client = this.clientRepository.findOneWithNotFoundDetection(clientId);
+
+            final String name = command.stringValueOfParameterNamed("name");
+            final String productType = command.stringValueOfParameterNamed("productType");
+            final BigDecimal maximumAmount = command.bigDecimalValueOfParameterNamed("maximumAmount");
+            final String dateFormat = command.dateFormat();
+            final String localeString = command.locale();
+            final Locale locale = Locale.forLanguageTag(localeString);
+            final LocalDate startDate = command.localDateValueOfParameterNamed("startDate");
+            final LocalDate endDate = command.localDateValueOfParameterNamed("endDate");
+
+
+            final LineOfCredit lineOfCredit = new LineOfCredit(client, name, productType, maximumAmount, startDate, endDate);
+            final LineOfCredit savedLineOfCredit = this.lineOfCreditRepository.save(lineOfCredit);
+
+
+            // Create a map with the line of credit data
+            final Map<String, Object> changes = new HashMap<>();
+            changes.put("resourceId", savedLineOfCredit.getId());
+            changes.put("clientId", savedLineOfCredit.getClient().getId());
+            changes.put("name", savedLineOfCredit.getName());
+            changes.put("productType", savedLineOfCredit.getProductType());
+            changes.put("maximumAmount", savedLineOfCredit.getMaximumAmount());
+            changes.put("availableBalance", savedLineOfCredit.getAvailableBalance());
+            changes.put("consumedAmount", savedLineOfCredit.getConsumedAmount());
```

Comment:
Fineract create operations does not expect changes, only updates expect changes. The only thing expected in the response is the resource Id